File: php/state_pattern/missionControl.php
<?php 

class MissionControl{
	public $state;
	public $systemsGo;
	public $mainEngine;
	public $fireRockets;
	public $missionAbort;
	public $liftOff ;

	public function __construct($tMinus){
	$this->systemsGo = new SystemsGo($this);
	$this->mainEngine = new MainEngine($this);
	$this->fireRockets = new FireRockets($this);
	$this->missionAbort = new MissionAbort($this);
	$this->liftOff = new LiftOff($this);
		if($tMinus>=60){
			echo 'Systems prepare for checks<br>';
			$this->state = new SystemsGo($this);
		}
		if($tMinus<60){
			echo 'Incorrect procedure. Mission abort.<br>';
			$this->state = new MissionAbort($this);
			$this->state->shutDown();
		}
	}

	public function setState(State $state){
		$this->state = $state;
	}

	public function liftOffHandOver($orbitState){
		echo $orbitState;
	}

	public function checkAllSystems(){
		$this->state->checkAllSystems();
	}

	public function startPowerUnits(){
		$this->state->startPowerUnits();
	}

	public function retractArms(){
		$this->state->retractArms();
	}

	public function rocketBooster(){
		$this->state->rocketBooster();
	}
public function shutDown(){
		$this->state->shutDown();
	}

	public function getStartPowerUnits(){
		return $this->mainEngine;
	}
	public function getRetractArms(){
		return $this->fireRockets;
	}

	public function getRocketBooster(){
		return $this->liftOff;
	}
	public function getAbortProcedure(){
		return "All systems are shutting down and fuel is being drained.<br>";
	}
	public function getLiftOffStatus(){
		return "We have a successful lift-Off.<br>";
	}

}

File: java/statePattern/MissionControl.java
package statePattern;

public class MissionControl {
	private State state;
	private State systemsGo;
	private State mainEngine;
	private State fireRockets;
	private State missionAbort;
	private State liftOff;

	public MissionControl(int tMinus) {
		systemsGo = new SystemsGo(this);
		mainEngine = new MainEngine(this);
		fireRockets = new FireRockets(this);
		missionAbort = new MissionAbort(this);
		liftOff = new LiftOff(this);
		if (tMinus >= 60) {
			System.out.println("Systems prepare for checks");
			state = new SystemsGo(this);
		}
		if (tMinus < 60) {
			System.out.println("Incorrect procedure. Mission abort.");
			state = new MissionAbort(this);
			state.shutDown();
		}
	}

	public void setState(State state) {
		this.state = state;
	}

	public void liftOffHandOver(String orbitState) {
		System.out.println(orbitState);
	}

	public void checkAllSystems() {
		this.state.checkAllSystems();
	}

	public void startPowerUnits() {
		this.state.startPowerUnits();
	}

	public void retractArms() {
		this.state.retractArms();
	}

	public void rocketBooster() {
		this.state.rocketBooster();
	}

	public void shutDown() {
		this.state.shutDown();
	}

	public State getStartPowerUnits() {
		return this.mainEngine;
	}

	public State getRetractArms() {
		return this.fireRockets;
	}

	public State getRocketBooster() {
		return this.liftOff;
	}

	public State getAbortProcedure() {
		return this.missionAbort;
	}

	public State getLiftOffStatus() {
		return this.systemsGo;
	}

}
